src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java

```diff
@@ -203,13 +247,31 @@ public void mouseClicked(java.awt.event.MouseEvent event) {
                     return;
 
                 DefaultMutableTreeNode node = (DefaultMutableTreeNode)selectedPath.getLastPathComponent();
-                if (!node.isLeaf())
+                if (!node.isLeaf() || !(node.getUserObject() instanceof EntityNode)) {
```

Comment:
Why was this needed? Did you actually reach a case where a leaf wasn't an `EntityNode`?

Reply:
I added to avoid any problem in the cursor, I didn't actually reach a case where a leaf wasn't an `EntityNode`, but I was being prepared if something happened. What do you think?